Docs: Added ChildComponents to API list & Fixed NavigationFooter

Garderoben opened the pull request:
- Added possibility to add ChildComponents so they display on the docs pages.
- Moved NavigationFooter from MainLayout to DocsPage to solve the MaxWidth problem.
- Removed empty sections from API page.

I thought allot about how to add them, and ended up with this because if a name change we should get an error, eg contra using a string list or manually adding them per docs page.

I have not added every ChildComponent/Relative Component because for DatePicker as example their API page show the Pickers property's as well.

Known issues: If you are on one API page that have all sections and navigate to one that has not the section content navigation still lists them.

![ApiLinks](https://user-images.githubusercontent.com/10367109/125791295-34232d30-abb6-492c-9d81-dd19363dd533.gif)
![image](https://user-images.githubusercontent.com/10367109/125790209-27dd3a59-668d-47ad-9b9c-ef03779de181.png)

Closes #1660

JonBunator commented:
Closes #1660

henon commented on src/MudBlazor.Docs/Services/MenuService.cs:
```diff
@@ -30,30 +31,30 @@ public class MenuService : IMenuService
-            .AddItem("Menu", typeof(MudMenu))
+            .AddItem("Menu", typeof(MudMenu), new List<Type>() { typeof(MudMenuItem)})
             .AddItem("Message Box", typeof(MudMessageBox))
-            .AddItem("Nav Menu", typeof(MudNavMenu))
-            .AddItem("Tabs", typeof(MudTabs))
-            .AddItem("Progress", typeof(MudProgressCircular))
-            .AddItem("Dialog", typeof(MudDialog))
+            .AddItem("Nav Menu", typeof(MudNavMenu), new List<Type>() { typeof(MudNavLink), typeof(MudNavGroup) })
+            .AddItem("Tabs", typeof(MudTabs), new List<Type>() { typeof(MudTabPanel), typeof(MudDynamicTabs) })
+            .AddItem("Progress", typeof(MudProgressCircular), new List<Type>() { typeof(MudProgressLinear) })
+            .AddItem("Dialog", typeof(MudDialog), new List<Type>() { typeof(MudDialogInstance), typeof(MudDialogProvider) })
```
Here is a suggestion for brevity: you could make it a `Type[]` instead of `List<Type>` then you can reduce to:

```c#
.AddItem("Drawer", typeof(MudDrawer), new[] { typeof(MudDrawerHeader), typeof(MudDrawerContainer) })
```

henon replied:
even better, when you make it a `params Type[]` you can reduce to

```c#
.AddItem("Drawer", typeof(MudDrawer), typeof(MudDrawerHeader), typeof(MudDrawerContainer) )
```

Garderoben replied:
Oh yes this is allot better! will change it.

Garderoben replied:
This is changed now, so much better!

Garderoben commented:
There is one issue i have trouble solving, if you navigate to a childcomponent's api page i somehow need to get that childcomponents parents Name + Link for the back to page button.

JonBunator commented:
> 
> 
> There is one issue i have trouble solving, if you navigate to a childcomponent's api page i somehow need to get that childcomponents parents Name + Link for the back to page button.

Maybe you could change the url from:
`
https://mudblazor.com/components/listitem
`
to
`
https://mudblazor.com/components/list/listitem
`
I am not sure if this is easily possible

Garderoben commented:
> > There is one issue i have trouble solving, if you navigate to a childcomponent's api page i somehow need to get that childcomponents parents Name + Link for the back to page button.
> 
> Maybe you could change the url from:
> `https://mudblazor.com/api/listitem`
> to
> `https://mudblazor.com/api/list/listitem`
> I am not sure if this is easily possible

Its possible, i was thinking about that as well as "best" solution it just requires a little more work but not a problem.
Unless someone else have a better one?

henon commented:
yay, it builds.

Yomodo commented:
![image](https://user-images.githubusercontent.com/10358198/125852249-d7ef942b-2e70-468a-87ae-b117fc9a522b.png)

Garderoben commented:
Updated now, thanks to @henon and @just-the-benno 

- Now linked ChildComponents API pages link back to the Documented Components docs page.
- Fixed broken Canonical Link
- Moved ResizeListner to OnAfterRender for MudPortal becuse project would not run for me, @porkopek knows about it and aproved that i did the change.

I have tested as much as i can and could not find any issues anymore, so for me this can be merged unless you find something.

![Updated](https://user-images.githubusercontent.com/10367109/125959531-aaaba5dc-f416-4938-a21f-671d5d8ce1d9.gif)

mikes-gh commented:
@Garderoben For me its not natural to have to navigate the component and subcomponent api  through the examples page.  You could have one link on the examples page to the component and be able to navigate directly between the component and subcomponent api once there.

Garderoben commented:
> @Garderoben For me its not natural to have to navigate the component and subcomponent api  through the examples page.  You could have one link on the examples page to the component and be able to navigate directly between the component and subcomponent api once there.

It's good idea, i will add same child components to the API page.

Garderoben commented:
By the way, I will not add anymore features in this PR, like linking to other api pages from the api page itself as discussed above.
I think this is good enough for now and inline with other major component library's. I need to spend time on some other projects for MudBlazor now.

mikes-gh commented:
Sure its your repo 👍
